### DISCUSSION TITLE
side effect in reaction is alreay an action? It doesn't need to be wrapped in action?

### DISCUSSION THREAD

**Discussion by ygnoh** [mobxjs/mobx#3032] [Q&A] [answered]
Hello

Let's say `a`, `b` are `observable`s. So I think the side effect in the following code need to be wrapped in `action` because it modifies an `observable` variable `b`.

```js
reaction(
    () => a,
    () => { ...modify b... }
);
```

However there is no error even if I don't wrap the function in `action`. Is it because side effect is already an action or just a MobX bug?

**Comment by urugator** [comment, marked as answer]
Yes, see https://github.com/mobxjs/mobx/issues/2718
